### ISSUE TITLE
What method is used in APSIM NG for simulating competition in intercrops?

### ISSUE THREAD

**Issue by ElsaLagerquist** [APSIMInitiative/ApsimX#8545]
### What is your question?

Hi!
I'm working on a simulation of intercrops and want to explain how the model simulates uptake of water and nitrogen when two crops are grown simultaneously. To my understanding this is done by alternating which day each crop utilise a resource, which seems a bit strange since you get a change in the outputs of water and nitrogen uptake every day. Lately I heard APSIM NG should use a Runge-Kutta solution for this type of simulation, but I cannot find any information about this. (Maybe I'm not searching for the right terms.) Is this the case, and if so, how many timesteps per day is used when running the simulation?

Kind regards,
Elsa

**Comment by hol353** [comment]
>  To my understanding this is done by alternating which day each crop utilise a resource

This is the APSIM 7.10 approach, not APSIM NG. Intercropping in NG is very different. For soil water and nitrogen arbitration between competing crop models a Runge-Kutta approach is used by the SoilArbitrator model. Some very poor documentation can be found here: https://builds.apsim.info/api/nextgen/docs/SoilArbitrator.pdf. This doc needs a description of the approach used. Perhaps someone with a better understanding of how SoilArbitrator works can respond to this issue (and fix the doc).

Arbitration of light between competing crop models is done by the MicroClimate model. A detailed description of the model can be found here: https://www.apsim.info/wp-content/uploads/2019/09/Micromet.pdf. This document was created for APSIM 7.10 (but not used for intercropping in 7.10) but is still relevant for NG.

**Comment by peter-devoil** [comment]
The "live" documentation has a bit more information than the above pdfs:
![image](https://github.com/APSIMInitiative/ApsimX/assets/1924386/de25bd07-cd88-48cf-8096-56f0cff019d7)

But it would be good to know where all the references got to..

**Comment by hut104** [comment]
Yep, we use a single fourth order Runge-Kutta at a daily timestep to solve for all water uptakes for all plant roots in all layers and all zones.
